src/VisualStudio/Core/Def/Implementation/Progression/GraphQueries/SearchGraphQuery.cs

```diff
@@ -15,11 +16,29 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.Progression
 {
     internal sealed class SearchGraphQuery : IGraphQuery
     {
-        private readonly string _searchPattern;
+        private static readonly char[] DotCharacterArray = new[] { '.' };
+
+        private readonly string _lastPatternPart;
+        private readonly string _entireSearchPattern;
 
         public SearchGraphQuery(string searchPattern)
         {
-            _searchPattern = searchPattern;
+            _entireSearchPattern = searchPattern;
+
+            // If the pattern has a dot in it, then when we search the compilations, we actually
+            // want to only filter by the last name portion of hte pattern (as the compilers 
```

Comment:
hte -> the?  Can I recommend https://visualstudiogallery.msdn.microsoft.com/7c8341f1-ebac-40c8-92c2-476db8d523ce?